saddle-core/src/main/scala/org/saddle/array/package.scala

```diff
@@ -611,7 +611,7 @@ package object array {
 
   /** Flatten a sequence of arrays into a single array
     */
-  def flatten[@spec(Boolean, Int, Long, Double) T: ST](
+  def flatten[@spec T: CLM](
```

Comment:
Does this introduce a binary incompat? I suspect so.

Reply:
Yes, but even the first commit did so, therefore I took the liberty to go a bit further.

I don't think it is possible to find a good solution with the issues of ScalarTag without binary incompatibilities. 

I regard unnecessary boxing as a bug: it is the difference between the coding running or not running on commodity hardware for any realistic data volumes.

So for the above reasons I think this PR should be binary incompatible and increase a major version bump.

Reply:
> I don't think it is possible to find a good solution with the issues of ScalarTag without binary incompatibilities

The changes in the first commit looked binary compatible. The changes were mostly method body changes and default implementations moved up and down in the class hierarchy. I'd like to double check this.

> I regard unnecessary boxing as a bug: it is the difference between the program running or not running on commodity hardware for any realistic data volumes.

Isn't it possible to add specialization for the remaining primitive types in a backward compatible way? 

I think binary breakage should be avoided as much as possible. Rapid major versions is large cost on the user and saddle had one  ~3 months ago. If there is no alternative, I would recommend milestone releases, and take the opportunity to review other parts of the lib.

Reply:
Oh I didn't read the comments below before the answer hereabove. You already answered most my questions.